### PR TITLE
refactor: prep for new livepatch support status of "kernel upgrade required"

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Test Steps

No functionality change, all tests should still pass

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Review by lucasmoura — COMMENTED** [review]
I like the idea, but maybe we can create a new python package named `entitlements_info`  and call the module `livepatch_info` instead, just to make it simpler to understand what is the difference between the `livepatch` module on `entitlements` and this one

**Comment by orndorffgrant** [comment]
That's a good point @lucasmoura - I agree that both modules having the same name is confusing. 

My intent was to treat the new module similar to how we treat the `apt` or `snap` modules - as our collection of opinionated wrapper functions for calling that respective API. So e.g. any function that wraps a `subp` to `canonical-livepatch` would go in the new one.

And then the old livepatch module is all the logic that goes into managing the entitlement for livepatch.

So I'm not sure if `entitlement_info.livepatch_info` quite captures that (but could be convinced).

What if I renamed the new module to `livepatch_cli` and left it in it's current location. So it would be `from uaclient import livepatch_cli; livepatch_cli.status()` - would that make more sense?

**Comment by orndorffgrant** [comment]
Per out of band conversation: we decided to leave the module naming as it is for now. If we decide it is to unwieldy in the future then we can change it at that time.

File: uaclient/livepatch.py [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+HTTPS_PROXY_OPTION = "https-proxy"
+
+LIVEPATCH_CMD = "/snap/bin/canonical-livepatch"
+
+LIVEPATCH_API_V1_KERNELS_SUPPORTED = "/v1/api/kernels/supported"
+
+event = event_logger.get_event_logger()
+
+
+class LivepatchStatusStatusLivepatchFix(DataObject):
```
Comment (lucasmoura) [inline uaclient/livepatch.py:36]: Can we use `LivepatchPatchFixStatus` instead ?

File: uaclient/livepatch.py [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+    def __init__(
+        self,
+        status: Optional[List[LivepatchStatusStatus]],
+    ):
+        self.status = status
+
+
+def status() -> Optional[LivepatchStatusStatus]:
+    if not is_livepatch_installed():
+        logging.debug("canonical-livepatch is not installed")
```
[the remark this answers is not in the document]
Comment (orndorffgrant) [reply uaclient/livepatch.py:118]: Per out-of-band conversation: we won't move log messages to `messages` for now at least

**Review by lucasmoura — APPROVED** [review]

**Review by renanrodrigo — APPROVED** [review]

**Pull request merged**